backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/outputGenerators/DocxUtils.java

```diff
@@ -73,29 +76,37 @@ private static void styleTable(XWPFTable table) {
     public static void fillReleasesTable(XWPFTable table, Collection<LicenseInfoParsingResult> projectLicenseInfoResults) {
 
         for (LicenseInfoParsingResult result : projectLicenseInfoResults) {
-            Set<String> copyrights = Collections.EMPTY_SET;
-            Set<LicenseNameWithText> licenseNamesWithTexts = Collections.EMPTY_SET;
-            Set<String> acknowledgements = Collections.EMPTY_SET;
-            if (result.isSetLicenseInfo()) {
-                LicenseInfo licenseInfo = result.getLicenseInfo();
-                if (licenseInfo.isSetCopyrights()) {
-                    copyrights = licenseInfo.getCopyrights();
-                }
-                if (licenseInfo.isSetLicenseNamesWithTexts()) {
-                    licenseNamesWithTexts = licenseInfo.getLicenseNamesWithTexts();
-                    acknowledgements = licenseNamesWithTexts.stream()
-                            .map(LicenseNameWithText::getAcknowledgements)
-                            .filter(Objects::nonNull).collect(Collectors.toSet());
-                }
-            }
             String releaseName = nullToEmptyString(result.getName());
             String version = nullToEmptyString(result.getVersion());
+            if (result.getStatus()== LicenseInfoRequestStatus.SUCCESS) {
+                Set<String> copyrights = Collections.emptySet();
+                Set<LicenseNameWithText> licenseNamesWithTexts = Collections.emptySet();
+                Set<String> acknowledgements = Collections.emptySet();
+                if (result.isSetLicenseInfo()) {
+                    LicenseInfo licenseInfo = result.getLicenseInfo();
+                    if (licenseInfo.isSetCopyrights()) {
+                        copyrights = licenseInfo.getCopyrights();
+                    }
+                    if (licenseInfo.isSetLicenseNamesWithTexts()) {
+                        licenseNamesWithTexts = licenseInfo.getLicenseNamesWithTexts();
+                        acknowledgements = licenseNamesWithTexts.stream()
+                                .map(LicenseNameWithText::getAcknowledgements)
+                                .filter(Objects::nonNull).collect(Collectors.toSet());
+                    }
+                }
 
-            addTableRow(table, releaseName, version, licenseNamesWithTexts, acknowledgements, copyrights);
+                addReleaseTableRow(table, releaseName, version, licenseNamesWithTexts, acknowledgements, copyrights);
+            } else {
+                String filename = Optional.ofNullable(result.getLicenseInfo())
+                        .map(LicenseInfo::getFilenames)
+                        .map(l -> l.stream().findFirst().orElse(null))
```

Comment:
the line `.map(l -> l.stream().findFirst().orElse(null))` might be equivalent to `.flatMap(l -> l.stream().findFirst())`.

Reply:
yes, it is